src/dbnode/encoding/m3tsz/timestamp_encoder.go

```diff
@@ -182,6 +194,9 @@ func (enc *TimestampEncoder) writeAnnotation(stream encoding.OStream, ant ts.Ann
 	stream.WriteBytes(buf[:annotationLength])
 	stream.WriteBytes(ant)
 
+	if enc.PrevAnnotationChecksum != emptyAnnotationChecksum {
+		enc.metrics.annotationRewritten.Inc(1)
```

Comment:
Just add a comment (as we discussed offline) to explain that this is based on the expectation that normally the annotation would not get rewritten, and might be too expensive to keep this around if this expectation turns out to be false someday.

Reply:
https://github.com/m3db/m3/pull/3565/commits/61f0826830f3386f4b2d70344472c29496d5be36